tests/binary_rpc/binary_rpc.cpp

```diff
@@ -0,0 +1,48 @@
+// Glaze Library
+// For the license information refer to glaze.hpp
+
+#ifndef BOOST_UT_DISABLE_MODULE
+#define BOOST_UT_DISABLE_MODULE
+#endif
+
+#include "boost/ut.hpp"
+#include "glaze/binary/read.hpp"
+#include "glaze/binary/write.hpp"
+#include "glaze/binary/rpc.hpp"
+
+using namespace boost::ut;
+
+struct my_request_data
+{
+   uint32_t integer{};
+   std::string string{};
+   
+   struct glaze {
+      using T = my_request_data;
+      static constexpr auto value = glz::object(&T::integer, &T::string);
+   };
+};
+
+suite repe_write_read = [] {
+   "repe_write_read"_test = [] {
+      // This syntax works, but we can wrap this into a cleaner interface. Let this be the low level syntax.
+      my_request_data params{55, "hello"};
+      glz::repe::message<my_request_data> msg{glz::repe::header{.method = "func"}, params};
+      
+      std::string buffer;
+      glz::write_binary(msg, buffer);
+      
+      params = {};
+      msg.header = {};
+      expect(!glz::read_binary(msg, buffer));
```

Comment:
Can read_binary return an error with `partial` as an error code? In terms of networking, so we would know to `async_receive` more data into the buffer before trying again. 

Secondly would it be beneficial to be able to only read the header?
```cpp
glz::repe::header header{};
glz::read_binary(header, buffer);
```